mne/decoding/search_light.py

```diff
@@ -30,10 +33,24 @@ def __repr__(self):
             repr_str += ', fitted with %i estimators' % len(self.estimators_)
         return repr_str + '>'
 
-    def __init__(self, base_estimator, n_jobs=1):
+    def __init__(self, base_estimator, scoring=None, n_jobs=1):
+        from sklearn.metrics import make_scorer, get_scorer
+
         _check_estimator(base_estimator)
         self.base_estimator = base_estimator
         self.n_jobs = n_jobs
+        self.scoring = scoring
+
+        # If scoring is None (default), the predictions are internally
+        # generated by estimator.score(). Else, we must first get the
+        # predictions based on the scorer.
+        if not isinstance(self.scoring, str):
+            self.scoring = (make_scorer(self.scoring) if self.scoring is
+                            not None else self.scoring)
+
+        elif self.scoring is not None:
+            self.scoring = get_scorer(self.scoring)
+
```

Comment:
Don't change the param at init, this will break the cloning behavior.

get_scorer and make_scorer should be called at .score() not init